astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,240 @@
+
+Copy the below code for the ChangeBank application into  `src/components/LoginButton.tsx`.
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/LoginButton.tsx"}
+  lang="ts" />
+
+### Layout
+
+<Aside type="tip">
+If this is your first time using Next.js's App Router, you should read through <a href="https://nextjs.org/docs/app/building-your-application/routing" target="_blank">Routing Fundamentals</a>.
+</Aside>
+
+Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).
+
+This has the overall structure of our application. The other pages will be added where `{children}` is located.
+
+Create a new file named `src/app/layout.tsx` and copy the below code to create the layout for the ChangeBank application.
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/layout.tsx"}
+  lang="ts" />
+
+### Home Page
+
+Create a new file `src/app/page.tsx` which will have the Homepage details. Not much here just an image and another Login button.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/page.tsx"}
+  lang="ts" />
+
+
+### Account Page
+
+Create a new file `src/app/account/page.tsx` which will have the Account details.
+
+One special note here is that there is a check to see if the session is missing. If it is, you redirect back to the homepage which protects this page on the server for unauthorized access. (You can find the same when a user is logged in on the homepage, it will redirect to `/account`)
+
+Here's the contents of `src/app/account/page.tsx`.
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/account/page.tsx"}
+  lang="ts" />
+
+### Make Change Page
+
+Finally, we'll add some business logic for logged in users to make change with the following code in `src/app/makechange/page.tsx`:
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/makechange/page.tsx"}
+  lang="ts" />
+
+If the user session is not present the user is redirect back the the homepage at the base route. If the user is present then the `MakeChangeForm` is presented. Create a new file located at `/src/components/MakeChangeForm.tsx` with the below code. This component has all of the business logic needed for taking in a dollar amount of money and returning the correct amount of each coin.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/MakeChangeForm.tsx"}
+  lang="ts" />
+
+### Run the Application
+You can now open up an incognito window and visit the NextJS app at http://localhost:3000/. Log in with the user account you created when setting up FusionAuth, and you’ll see the email of the user next to a logout button.
+
+```shell
+npm run dev
```

Comment:
I get this error message when running through the instructions.

```
Module not found: Can't resolve '../components/LoginLink'
  3 | import { redirect } from 'next/navigation';
  4 | import { authOptions } from './api/auth/[...nextauth]/route';
> 5 | import LoginLink from '../components/LoginLink';
  6 | 
  7 | export default async function Home() {
  8 |   const session = await getServerSession(authOptions);

https://nextjs.org/docs/messages/module-not-found
```

Looks like we're missing installing a `LoginLink` component. Can you add that step in, please?

```
dan@mooreds fusionauth-quickstart-javascript-nextjs-web % ll complete-application/src/components 
total 24
-rw-r--r--  1 dan  staff   482 Sep 21 13:32 LoginButton.tsx
-rw-r--r--  1 dan  staff   390 Sep 21 13:32 LoginLink.tsx
-rw-r--r--  1 dan  staff  1904 Sep 21 13:32 MakeChangeForm.tsx
dan@mooreds fusionauth-quickstart-javascript-nextjs-web % ll changebank/src/components        
total 16
-rw-r--r--  1 dan  staff   482 Sep 21 13:44 LoginButton.tsx
-rw-r--r--  1 dan  staff  1904 Sep 21 13:46 MakeChangeForm.tsx

```